### PR TITLE
deploy an application

### PR DESCRIPTION
**Pull request by w8896699**

### CODE REVIEW

**Comment by Pilargit12** [comment]
For reviewers, can you please (if you think of anything that could help enhance the Related Pages from this document, please add those in its corresponding section of the document. Thank you for your help!

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+## Configuring deployment
+
+### 1. Understanding deployment configurations and their components
+
+Deployment Configurations  in OpenShift are a recipe for deploying your application. They consist of the following are considered as most important field:
+
+- **Pod Template:** This defines the desired state of the pods being deployed, including the base image, ports to expose, and environment variables
+- **Replicas:** The number of instances of your application that should be maintained
+- **Selectors:** Labels that identify the pods managed by the deployment configuration
+- **Triggers:** Events that will cause a new deployment, such as a new image build
```
Comment (ShellyXueHan) [inline src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:127]: better to be clear that only dc and bc have triggers, deployment and sts do not

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+Understanding these components is essential for correctly configuring your application's deployment. To read more about other configuration options, use `oc explain` command or this [kubernetes official documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).
+
+### 2. Defining deployment strategies
+
+OpenShift supports several deployment strategies, including:
```
Comment (ShellyXueHan) [inline src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:133]: better to be clear that via the configuration, only rolling and recreate are available options, blue-green and canary need to be setup explicitly via image tags, labels and different routes.

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+We explored the essential aspects of data storage management in OpenShift and emphasized the significance of securing your deployed applications. Each of these steps plays a pivotal role in achieving efficient deployment and operation within the OpenShift environment.
+
+It's important to note that this guide only provides an introduction to the vast range of capabilities OpenShift has to offer. As OpenShift continues to evolve,  new features will be introduced, expanding the possibilities even further. The more you engage with OpenShift, the more confident and innovative you'll become in deploying and managing your applications. 
+
+We highly encourage you to dive deeper, experiment with new concepts, and continuously explore the vast capabilities of OpenShift. To expand your knowledge and stay updated with the latest best practices and features, we recommend regularly referring to the official OpenShift and the B.C. government developer [documentation](https://developer.gov.bc.ca/), as it is a great resource to expand your knowledge and stay up-to-date with the latest best practices and features.
```
Comment (ShellyXueHan) [inline src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:333]: do we still want to direct ppl to devhub?

**Review by ShellyXueHan — COMMENTED** [review]
great work!

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+For guidance on resource tuning, refer to this [documentation](https://docs.developer.gov.bc.ca/application-resource-tuning/). You can also watch this [video](https://www.youtube.com/watch?v=rkxVZgn9icU&t=14s) for more explanation. Understanding and adjusting these resources according to your application's needs is a critical aspect of deployment preparation.
+
+---
+
+## Deploying the application
+
+### 1. Creating a deployment using OpenShift CLI or Web Console
+
+To deploy an application, you can use the OpenShift CLI (`oc create-app`) or the web console. Both offer the same capabilities and your choice will be dependent on personal preference. More information can be found in the [official OpenShift documentation](https://docs.openshift.com/container-platform/latest/applications/application_life_cycle_management/creating-applications-using-cli.html).
```
[the remark this answers is not in the document]
Comment (Pilargit12) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:100]: @w8896699 can we get consensus on this suggested change? Thank you so much :-)

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+
+### 2. Configuring resource limits and scaling options
+
+Resource limits and scaling options are an important part of managing application performance and availability. 
+
+OpenShift allows you to set resource limits (CPU and memory) for each Pod in your deployments, which can prevent a Pod from using more resources than necessary.
```
[the remark this answers is not in the document]
Comment (Pilargit12) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:106]: @w8896699 same for this one? Thank you so much

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md (hunk 3 above)
Comment (Pilargit12) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:333]: At the moment, nothing has changed.  So we shall leave it there if this is a used resourced now, what do you think? @w8896699

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (w8896699) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:100]: Yes, it should be new-app. I got it wrong

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (w8896699) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:106]: Yes it should be each container.

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md (hunk 1 above)
Comment (StevenBarre) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:127]: @ShellyXueHan Deployments and STS can have image triggers, its just a bit more complicated https://docs.openshift.com/container-platform/4.12/openshift_images/triggering-updates-on-imagestream-changes.html

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+
+OpenShift seamlessly integrates with popular version control systems, such as Git, enabling effective version tracking and triggering deployments upon codebase updates. Additionally, OpenShift leverages image tags to monitor different versions of container images.
+
+Image tags offer a flexible reference to images that can be modified over time, facilitating efficient implementation of versioning strategies, including semantic versioning.
+
+Nevertheless, it is crucial to exercise caution  when managing mutable tags. For detailed insights into best practices for image stream and tag management in OpenShift, please refer to the documentation on [best practices for manageing image streams](https://docs.developer.gov.bc.ca/best-practices-for-managing-image-streams/) .
+
+### 3. Backup and Restoration in a CD Context
+
+Regularly backing up your application data is crucial for safeguarding against potential risks. In a CD context, you can automate and integrate backup processes into your deployment pipeline, ensuring that backups stay up to date with any changes made.
```
Comment (StevenBarre) [inline src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:267]: Add link about backups
```suggestion
Regularly [backing up your application data](https://docs.developer.gov.bc.ca/recovery-responsibilities/) is crucial for safeguarding against potential risks. In a CD context, you can automate and integrate backup processes into your deployment pipeline, ensuring that backups stay up to date with any changes made. 
```

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+
+A comprehensive guide on how to use PVCs in your deployments can be found on the [platform storage documentation](https://docs.developer.gov.bc.ca/platform-storage/).
+
+### 3. Backing Up and Restoring Data
+
+Data loss is a critical issue that needs attention in any environment, including OpenShift. To address this concern, the B.C. government has developed the  `backup-container` application specifically for backing up and restoring data within OpenShift.   It's designed to provide a consistent way to back up the state of applications running in OpenShift. 
+
+You can learn more about this from the [backup-container GitHub page](https://github.com/bcgov/backup-container-compliance-enforcement). Also refer to the disastory recover documentation [Persistent volumes section](https://github.com/bcgov/platform-developer-docs/blob/main/src/docs/automation-and-resiliency/namespace-recovery-and-responsibilities.md#persistent-volumes) that we mentioned eariler.
```
Comment (StevenBarre) [inline src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:291]: Fix link
```suggestion
You can learn more about this from the [backup-container GitHub page](https://github.com/bcgov/backup-container-compliance-enforcement). Also refer to the disastory recover documentation [Persistent volumes section](https://docs.developer.gov.bc.ca/recovery-responsibilities/#persistent-volumes) that we mentioned eariler. 
```

File: src/docs/build-deploy-and-maintain-apps/deploy-an-application.md (hunk 3 above)
Comment (w8896699) [reply src/docs/build-deploy-and-maintain-apps/deploy-an-application.md:333]: I don't know what doc site we need to refer now..we have too many of them

**Review by Pilargit12 — COMMENTED** [review]
Perfect it's ready to merge to main! Thanks everyone for your help!

**Pull request merged**